lib/clients/reports.ex

```diff
@@ -0,0 +1,44 @@
+defmodule GlassFactoryApi.Clients.Reports do
+  alias GlassFactoryApi.ApiClient
+  alias GlassFactoryApi.Clients.Reports.TimeReport
+
+  @doc """
+  List time reports from the given client id
+
+  ## Examples
+
+      iex> GlassFactoryApi.Clients.Reports(2079)
+      [ %TimeReport{
+          client_id: 2079,
+          project_id: 14330,
+          job_id: nil,
+          activity_id: nil,
+          user_id: 3,
+          role_id: 1480,
+          date: "2018-07-10",
+          planned: 8,
+          time: 0
+        }
+      ]
+  """
+  @spec list_time_reports(integer(), map()) :: {:error, String.t()} | {:ok, [TimeReport]}
+  def list_time_reports(client_id, config \\ %{}) do
+    client_request = ApiClient.get("clients/#{client_id}/reports/time", config)
+
+    with {:ok, %{status: 200, body: body}} <- client_request do
+      time_reports = Enum.map(body, &TimeReport.to_struct(&1))
+
+      {:ok, time_reports}
+    else
+      {:ok, %{status: 404}} -> {:error, "Can't find time reports for client id #{client_id}"}
```

Comment:
Here you can add some more patterns that handle when a request wasn't successful when the internet is not available for example:

``` elixir
#  When internet isn't available Tesla returns us the following
ApiClient.get("your_url", config)
{:error , :econnrefused}

# Then you could do something like the following to handle this type of error
{:error, error} when is_atom(error) -> {:error, Atom.to_string(error)} 
```

And the tuple with `:error` will be returned 😄

Reply:
In time, this kind of error could be in an upper-level handler, I'll create an issue so we can discuss it 😄

Reply:
Nice point @Cendrao!, we've decided at Slack (me and @Cendrao) that we will handle it on `ApiClient` in another PR 😄